#2179 - Added metadata only mode for retrieving a modern page

estruyf opened the pull request:
Added an extra flag `--metadataOnly` flag to skip the control and section parsing which fails to depend on the contents of the web parts and sections.

Here is an example:

![Screenshot 2021-02-17 at 20 48 38](https://user-images.githubusercontent.com/2900833/108259808-d7ba2880-7161-11eb-877d-6ab3411bc120.png)

Top is when using the `--metadataOnly` flag, bottom without it.

waldekmastykarz commented:
That seems to me like a workaround though. If there is an issue with parsing `*` we should address that rather than introducing a new flag that works around the problem. It seems like only a matter of time before we'll need to handle `*` in the content, so let's fix it right away.

waldekmastykarz commented:
We discussed it separately in more detail and arrived to the conclusion that this addition has its place, and we still need to fix the issue with parsing content with `*` which will be done separately.

estruyf commented:
Ok @waldekmastykarz, like discussed, I updated the `spo page get` command with the extra API call. I also updated all tests to make them easier readable by using mocks only.

waldekmastykarz commented:
Awesome! Thank you! Let's get this in 👏

waldekmastykarz reviewed (approved):
Awesome! It's in 👏

waldekmastykarz commented:
Merged manually. Thank you! 👏
